pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go

```diff
@@ -367,6 +395,64 @@ runbook: https://github.com/openshift/ops-sop/blob/master/v4/alerts/cluster_has_
 	return nil
 }
 
+// snitchResourcesExist checks if the associated cluster resources for a snitch exist
+func (r *ReconcileDeadmansSnitchIntegration) snitchResourcesExist(dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration, cd *hivev1.ClusterDeployment) (bool, error) {
+	logger := log.WithValues("DeadMansSnitchIntegration.Namespace", dmsi.Namespace, "DMSI.Name", dmsi.Name, "cluster-deployment.Name:", cd.Name, "cluster-deployment.Namespace:", cd.Namespace)
+
+	dmsSecret := utils.SecretName(cd.Spec.ClusterName, dmsi.Spec.SnitchNamePostFix)
+	logger.Info("Checking if secret exists")
+	err := r.client.Get(context.TODO(),
+		types.NamespacedName{Name: dmsSecret, Namespace: cd.Namespace},
+		&corev1.Secret{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+
+	logger.Info("Checking if syncset exists")
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: dmsSecret, Namespace: cd.Namespace}, &hivev1.SyncSet{})
```

Comment:
This will not be called if the secret doesn't exist and the result returned will be `false` indicating there is no DMS snitch to cleanup.  If secret OR syncset exist need to treat as though the snitch _does_ exist.  Else we have an inconsistent state for deletion.

Reply:
:disappointed: 
Thanks @jewzaam 
Changes are in https://github.com/openshift/deadmanssnitch-operator/pull/117/commits/cf9f7723f8b7cdd77fd694518fbd32ebfebb0bee
I have changed the signature a bit to handle delete if `secretExist || syncSetExist` and create if `!secretExist || !syncSetExist`
Not sure if there is nicer way to define write the logicg